pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1242,43 +1207,69 @@ func (vs *VSphere) CreateVolume(volumeOptions *VolumeOptions) (volumePath string
 		return "", err
 	}
 
-	// vmdks will be created inside kubevols directory
-	kubeVolsPath := filepath.Clean(ds.Path(VolDir)) + "/"
-	err = makeDirectoryInDatastore(vs.client, dc, kubeVolsPath, false)
-	if err != nil && err != ErrFileAlreadyExist {
-		glog.Errorf("Cannot create dir %#v. err %s", kubeVolsPath, err)
-		return "", err
-	}
-	glog.V(4).Infof("Created dir with path as %+q", kubeVolsPath)
+	// Create a disk with the VSAN storage capabilities specified in the volumeOptions.StorageProfileData.
+	// This is achieved by following steps:
+	// 1. Create dummy VM if not already present.
+	// 2. Reconfigure the VM with the disk.
+        // 3. Detach the disk from the dummy VM.
+	if volumeOptions.StorageProfileData != "" {
+		// Check if the datastore is VSAN if any capability requirements are specified.
+		// VSphere cloud provider now only supports VSAN capabilities requirements
+		ok, err := checkIfDatastoreTypeIsVSAN(vs.client, ds)
+		if err != nil {
+			return "", fmt.Errorf("Failed while determining whether the datastore: %q"+
+				" is VSAN or not.", datastore)
+		}
 
-	vmDiskPath := kubeVolsPath + volumeOptions.Name + ".vmdk"
+		if !ok {
+			return "", fmt.Errorf("The specified datastore: %q is not a VSAN datastore."+
+				" The policy parameters will work only with VSAN Datastore."+
+				" So, please specify a valid VSAN datastore", datastore)
+		}
 
-	// Create a virtual disk manager
-	virtualDiskManager := object.NewVirtualDiskManager(vs.client.Client)
+		// Check if the VM exists.
+		vmRegex := vs.cfg.Global.WorkingDir + DummyVMName
+		dummyVM, err := f.VirtualMachine(ctx, vmRegex)
+		if err != nil {
+			// 1. Create dummy VM.
+			err := vs.createDummyVM(ctx, dc, ds)
```

Comment:
Currently, if there is no storage policy specified, the disks would be created under kubevols folder. This is an already existing feature. Now, if we have all the disks stored under dummyVM directory, it would create confusion for the already existing users. 

So, what about the customers who are already using previous kubernetes releases have all VMDK's stored under kubevols. And now with the new release we create a new dummyVM folder for storing all the VMDK's. How do we migrate existing VMDK's from kubevols to this new dummyVM folder? Should users consider these 2 places for managing VMDK's? (Upgrade scenario).

Also, consider the following scenario where we modify code in all places to use dummyVM folder for storing newly created VMDK's as suggested by you. 
1. Create a storage class with no storage policy. We need a place to store with this VMDK. So, we create a new directory for dummyVM folder and place the VMDK in this.
2. Create a storage class with storage policy. There is already a folder with this name. I guess VC would create a new VM folder with dummyVM_folder_1 for storing VMDK's now. 

Again it creates confusion where exactly are the VMDK's.. 

So, its better to have VMDK's in 1 place which is inside the subfolder kubevols and that can be achieved with reconfigure.